### DISCUSSION TITLE
How do re-render/refresh the specific component or the entire canvas?

### DISCUSSION THREAD

**Discussion by iamqinglong** [GrapesJS/grapesjs#4973] [Q&A]
I conditionally load my plugin(lottery-entry) if the other plugin(lottery) exist then my lottery-entry plugin doesn't render in the canvas because of I think I load it late(but it needs to be late cuz it is conditional).
I tried `editor.refresh()` but it seems it doesn't work that way.

```
if (this.lotteryComponentExist()) {
    loadLotteryEntry(this.editor)
    this.editor.refresh()
}
```

**Comment by artf** [comment]
You can force the component to re-render in this way `component.trigger('rerender')`
